### ISSUE TITLE
resolve warnings during "-all.jar" jar generation

### ISSUE THREAD

**Issue by romani** [checkstyle/checkstyle#756]
mvn clean package -Passembly 

```
[WARNING] org.abego.treelayout.core-1.0.1.jar, antlr4-runtime-4.5.jar define 19 overlappping classes: 
[WARNING]   - org.abego.treelayout.util.DefaultConfiguration
[WARNING]   - org.abego.treelayout.TreeLayout
[WARNING]   - org.abego.treelayout.util.DefaultTreeForTreeLayout
[WARNING]   - org.abego.treelayout.internal.util.java.util.ListUtil
[WARNING]   - org.abego.treelayout.internal.util.java.util.IteratorUtil$ReverseIterator
[WARNING]   - org.abego.treelayout.internal.util.Contract
[WARNING]   - org.abego.treelayout.internal.util.java.lang.string.StringUtil
[WARNING]   - org.abego.treelayout.TreeLayout$DumpConfiguration
[WARNING]   - org.abego.treelayout.internal.util.java.lang.IterableUtil$ReverseIterable
[WARNING]   - org.abego.treelayout.NodeExtentProvider
[WARNING]   - 9 more...
[WARNING] maven-shade-plugin has detected that some .class files
[WARNING] are present in two or more JARs. When this happens, only
[WARNING] one single version of the class is copied in the uberjar.
[WARNING] Usually this is not harmful and you can skeep these
[WARNING] warnings, otherwise try to manually exclude artifacts
[WARNING] based on mvn dependency:tree -Ddetail=true and the above
[WARNING] output
[WARNING] See http://docs.codehaus.org/display/MAVENUSER/Shade+Plugin

```

Expected: no warnings.

**Comment by romani** [comment]
org.abego.treelayout.core is used in antlr and antlr4 artifacts, our main library is still 2.7.7, so version from 4.5 is excluded.
